lib/webauthn/authenticator_response.rb

```diff
@@ -1,18 +1,35 @@
 # frozen_string_literal: true
 
+require "webauthn/error"
+
 module WebAuthn
+  class VerificationError < Error; end
+
+  class AuthenticatorDataVerificationError < VerificationError; end
+  class ChallengeVerificationError < VerificationError; end
+  class OriginVerificationError < VerificationError; end
+  class RPIdVerificationError < VerificationError; end
+  class TypeVerificationError < VerificationError; end
+  class UserPresenceVerificationError < VerificationError; end
+
   class AuthenticatorResponse
     def initialize(client_data_json:)
       @client_data_json = client_data_json
     end
 
-    def valid?(original_challenge, original_origin, rp_id: nil)
+    def verify(original_challenge, original_origin, rp_id: nil)
       valid_type? &&
         valid_challenge?(original_challenge) &&
         valid_origin?(original_origin) &&
+        verify_authenticator_data &&
         valid_rp_id?(rp_id || rp_id_from_origin(original_origin)) &&
-        authenticator_data.valid? &&
-        authenticator_data.user_flagged?
+        verify_user_flagged
```

Comment:
This still smells a bit.
Some calls named `valid_*`, others `verify_*`...

Need to refactor.